pkg/command/lima.go

```diff
@@ -122,14 +127,38 @@ func (lcc *limaCmdCreator) create(stdin io.Reader, stdout, stderr io.Writer, arg
 	lcc.logger.Debugf("Creating limactl command: ARGUMENTS: %v, %s: %s", args, envKeyLimaHome, lcc.limaHomePath)
 	cmd := lcc.cmdCreator.Create(lcc.limactlPath, args...)
 	limaHomeEnv := fmt.Sprintf("%s=%s", envKeyLimaHome, lcc.limaHomePath)
-	path := lcc.systemDeps.Env(envKeyPath)
-	// TODO: Refactor this; don't need qemuBinPath for windows
-	path = fmt.Sprintf("%s:%s", lcc.qemuBinPath, path)
-	pathEnv := fmt.Sprintf("%s=%s", envKeyPath, path)
-	env := append(lcc.systemDeps.Environ(), limaHomeEnv, pathEnv)
-	cmd.SetEnv(env)
+	var pathEnv string
+	var envKeyPath string
+	var path string
+	if runtime.GOOS == "windows" {
+		envKeyPath = envKeyWinPath
+		path = lcc.systemDeps.Env(envKeyPath)
+		path = fmt.Sprintf(`%s\;%s`, lcc.binPath, path)
+		pathEnv = fmt.Sprintf("%s=%s", envKeyPath, path)
+	} else {
+		envKeyPath = envKeyUnixPath
+		path = lcc.systemDeps.Env(envKeyPath)
+		path = fmt.Sprintf("%s:%s", lcc.binPath, path)
+		pathEnv = fmt.Sprintf("%s=%s", envKeyPath, path)
+	}
+
+	newPathEnv := replaceOrAppend(lcc.systemDeps.Environ(), envKeyLimaHome, limaHomeEnv)
```

Comment:
Why do we need this? Doesn't line 141 ensure that we put lcc.binPath first, so even if there is another limactl in the path it wont be picked since lcc.binPath is first.

Reply:
The old code did:

```
env := append(lcc.systemDeps.Environ(), limaHomeEnv, pathEnv)
```,

so no matter what, the new path would actually be at the end

Reply:
The prepend code would get rid of the `replaceOrAppend` function and replace it with
```
	env := append([]string{limaHomeEnv, pathEnv}, lcc.systemDeps.Environ()...)
```

Testing to see if it actually works on Windows

Reply:
@vsiravar, hmm, that didn't work on Windows at least. Not sure how the Path rules work

Logs:
```
PS C:\Users\Administrator\Code\finch> ./_output/bin/finch.exe vm init
time="2023-10-12T21:12:41Z" level=info msg="newEnv: [LIMA_HOME=C:\\Users\\Administrator\\Code\\finch\\_output\\lima\\data Path=C:\\Users\\Administrator\\Code\\finch\\_output\\lima\\bin\\;C:\\Windows\\System32\\;C:\\Program Files\\WSL\\ ALLUSERSPROFILE=C:\\ProgramData APPDATA=C:\\Users\\Administrator\\AppData\\Roaming AWS_EXECUTION_ENV=EC2 CommonProgramFiles=C:\\Program Files\\Common Files CommonProgramFiles(x86)=C:\\Program Files (x86)\\Common Files CommonProgramW6432=C:\\Program Files\\Common Files COMPUTERNAME=EC2AMAZ-3QHKPN9 ComSpec=C:\\Windows\\system32\\cmd.exe DriverData=C:\\Windows\\System32\\Drivers\\DriverData EC2LAUNCH_TELEMETRY=1 FPS_BROWSER_APP_PROFILE_STRING=Internet Explorer FPS_BROWSER_USER_PROFILE_STRING=Default GOPATH=C:\\Users\\Administrator\\go HOMEDRIVE=C: HOMEPATH=\\Users\\Administrator LOCALAPPDATA=C:\\Users\\Administrator\\AppData\\Local LOGONSERVER=\\\\EC2AMAZ-3QHKPN9 NUMBER_OF_PROCESSORS=48 OS=Windows_NT Path=C:\\Windows\\System32\\;C:\\Program Files\\WSL\\ PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL POWERSHELL_DISTRIBUTION_CHANNEL=MSI:Windows Server 2022 Datacenter PROCESSOR_ARCHITECTURE=AMD64 PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 85 Stepping 7, GenuineIntel PROCESSOR_LEVEL=6 PROCESSOR_REVISION=5507 ProgramData=C:\\ProgramData ProgramFiles=C:\\Program Files ProgramFiles(x86)=C:\\Program Files (x86) ProgramW6432=C:\\Program Files PSModulePath=C:\\Users\\Administrator\\Documents\\PowerShell\\Modules;C:\\Program Files\\PowerShell\\Modules;c:\\program files\\powershell\\7\\Modules;C:\\Program Files\\WindowsPowerShell\\Modules;C:\\Windows\\system32\\WindowsPowerShell\\v1.0\\Modules;C:\\Program Files (x86)\\AWS Tools\\PowerShell\\ PUBLIC=C:\\Users\\Public SystemDrive=C: SystemRoot=C:\\Windows TEMP=C:\\Users\\ADMINI~1\\AppData\\Local\\Temp TMP=C:\\Users\\ADMINI~1\\AppData\\Local\\Temp USERDOMAIN=EC2AMAZ-3QHKPN9 USERDOMAIN_ROAMINGPROFILE=EC2AMAZ-3QHKPN9 USERNAME=Administrator USERPROFILE=C:\\Users\\Administrator windir=C:\\Windows WSLENV=WT_SESSION::WT_PROFILE_ID WT_PROFILE_ID={574e775e-4f2a-5b96-ac1e-a2962a402336} WT_SESSION=2a5ccf8c-1d0b-40b8-a291-6c07c5592132]"
time="2023-10-12T21:12:41Z" level=info msg="attaching disk at path: C:\\Users\\Administrator\\AppData\\Local\\.finch\\.disks\\976377f8dcd57082.vhdx"
time="2023-10-12T21:12:42Z" level=info msg="newEnv: [LIMA_HOME=C:\\Users\\Administrator\\Code\\finch\\_output\\lima\\data Path=C:\\Users\\Administrator\\Code\\finch\\_output\\lima\\bin\\;C:\\Windows\\System32\\;C:\\Program Files\\WSL\\ ALLUSERSPROFILE=C:\\ProgramData APPDATA=C:\\Users\\Administrator\\AppData\\Roaming AWS_EXECUTION_ENV=EC2 CommonProgramFiles=C:\\Program Files\\Common Files CommonProgramFiles(x86)=C:\\Program Files (x86)\\Common Files CommonProgramW6432=C:\\Program Files\\Common Files COMPUTERNAME=EC2AMAZ-3QHKPN9 ComSpec=C:\\Windows\\system32\\cmd.exe DriverData=C:\\Windows\\System32\\Drivers\\DriverData EC2LAUNCH_TELEMETRY=1 FPS_BROWSER_APP_PROFILE_STRING=Internet Explorer FPS_BROWSER_USER_PROFILE_STRING=Default GOPATH=C:\\Users\\Administrator\\go HOMEDRIVE=C: HOMEPATH=\\Users\\Administrator LOCALAPPDATA=C:\\Users\\Administrator\\AppData\\Local LOGONSERVER=\\\\EC2AMAZ-3QHKPN9 NUMBER_OF_PROCESSORS=48 OS=Windows_NT Path=C:\\Windows\\System32\\;C:\\Program Files\\WSL\\ PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL POWERSHELL_DISTRIBUTION_CHANNEL=MSI:Windows Server 2022 Datacenter PROCESSOR_ARCHITECTURE=AMD64 PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 85 Stepping 7, GenuineIntel PROCESSOR_LEVEL=6 PROCESSOR_REVISION=5507 ProgramData=C:\\ProgramData ProgramFiles=C:\\Program Files ProgramFiles(x86)=C:\\Program Files (x86) ProgramW6432=C:\\Program Files PSModulePath=C:\\Users\\Administrator\\Documents\\PowerShell\\Modules;C:\\Program Files\\PowerShell\\Modules;c:\\program files\\powershell\\7\\Modules;C:\\Program Files\\WindowsPowerShell\\Modules;C:\\Windows\\system32\\WindowsPowerShell\\v1.0\\Modules;C:\\Program Files (x86)\\AWS Tools\\PowerShell\\ PUBLIC=C:\\Users\\Public SystemDrive=C: SystemRoot=C:\\Windows TEMP=C:\\Users\\ADMINI~1\\AppData\\Local\\Temp TMP=C:\\Users\\ADMINI~1\\AppData\\Local\\Temp USERDOMAIN=EC2AMAZ-3QHKPN9 USERDOMAIN_ROAMINGPROFILE=EC2AMAZ-3QHKPN9 USERNAME=Administrator USERPROFILE=C:\\Users\\Administrator windir=C:\\Windows WSLENV=WT_SESSION::WT_PROFILE_ID WT_PROFILE_ID={574e775e-4f2a-5b96-ac1e-a2962a402336} WT_SESSION=2a5ccf8c-1d0b-40b8-a291-6c07c5592132]"
time="2023-10-12T21:12:42Z" level=info msg="Initializing and starting Finch virtual machine..."
time="2023-10-12T21:12:44Z" level=error msg="Finch virtual machine failed to start, debug logs:\ntime=\"2023-10-12T21:12:43Z\" level=info msg=\"Terminal is not available, proceeding without opening an editor\"\ntime=\"2023-10-12T21:12:43Z\" level=warning msg=\"expected status \\\"Stopped\\\", got \\\"Uninitialized\\\"\"\ntime=\"2023-10-12T21:12:43Z\" level=warning msg=\"Ignoring: vmType wsl2: [OS CPUs Memory Firmware Audio Video GuestInstallPrefix CACertificates Rosetta]\"\ntime=\"2023-10-12T21:12:43Z\" level=info msg=\"Attempting to download the nerdctl archive\" arch=x86_64 digest=\"sha256:2c5c43a8b77ed62090241027361baa62d8fb70a759bc9e7a82c637135598701f\" location=\"https://github.com/containerd/nerdctl/releases/download/v1.6.0/nerdctl-full-1.6.0-linux-amd64.tar.gz\"\ntime=\"2023-10-12T21:12:43Z\" level=info msg=\"Using cache \\\"C:\\\\\\\\Users\\\\\\\\Administrator\\\\\\\\AppData\\\\\\\\Local\\\\\\\\lima\\\\\\\\download\\\\\\\\by-url-sha256\\\\\\\\addb72512fd2fef72b08d8648173d7aaec3b4d5ae0566ac0bfd03c5adeeafc78\\\\\\\\data\\\"\"\ntime=\"2023-10-12T21:12:43Z\" level=info msg=\"[hostagent] time=\\\"2023-10-12T21:12:43Z\\\" level=info msg=\\\"path (from Lima): [LIMA_HOME=C:\\\\\\\\Users\\\\\\\\Administrator\\\\\\\\Code\\\\\\\\finch\\\\\\\\_output\\\\\\\\lima\\\\\\\\data ALLUSERSPROFILE=C:\\\\\\\\ProgramData APPDATA=C:\\\\\\\\Users\\\\\\\\Administrator\\\\\\\\AppData\\\\\\\\Roaming AWS_EXECUTION_ENV=EC2 CommonProgramFiles=C:\\\\\\\\Program Files\\\\\\\\Common Files CommonProgramFiles(x86)=C:\\\\\\\\Program Files (x86)\\\\\\\\Common Files CommonProgramW6432=C:\\\\\\\\Program Files\\\\\\\\Common Files COMPUTERNAME=EC2AMAZ-3QHKPN9 ComSpec=C:\\\\\\\\Windows\\\\\\\\system32\\\\\\\\cmd.exe DriverData=C:\\\\\\\\Windows\\\\\\\\System32\\\\\\\\Drivers\\\\\\\\DriverData EC2LAUNCH_TELEMETRY=1 FPS_BROWSER_APP_PROFILE_STRING=Internet Explorer FPS_BROWSER_USER_PROFILE_STRING=Default GOPATH=C:\\\\\\\\Users\\\\\\\\Administrator\\\\\\\\go HOMEDRIVE=C: HOMEPATH=\\\\\\\\Users\\\\\\\\Administrator LOCALAPPDATA=C:\\\\\\\\Users\\\\\\\\Administrator\\\\\\\\AppData\\\\\\\\Local LOGONSERVER=\\\\\\\\\\\\\\\\EC2AMAZ-3QHKPN9 NUMBER_OF_PROCESSORS=48 OS=Windows_NT Path=C:\\\\\\\\Windows\\\\\\\\System32\\\\\\\\;C:\\\\\\\\Program Files\\\\\\\\WSL\\\\\\\\ PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL POWERSHELL_DISTRIBUTION_CHANNEL=MSI:Windows Server 2022 Datacenter PROCESSOR_ARCHITECTURE=AMD64 PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 85 Stepping 7, GenuineIntel PROCESSOR_LEVEL=6 PROCESSOR_REVISION=5507 ProgramData=C:\\\\\\\\ProgramData ProgramFiles=C:\\\\\\\\Program Files ProgramFiles(x86)=C:\\\\\\\\Program Files (x86) ProgramW6432=C:\\\\\\\\Program Files PSModulePath=C:\\\\\\\\Users\\\\\\\\Administrator\\\\\\\\Documents\\\\\\\\PowerShell\\\\\\\\Modules;C:\\\\\\\\Program Files\\\\\\\\PowerShell\\\\\\\\Modules;c:\\\\\\\\program files\\\\\\\\powershell\\\\\\\\7\\\\\\\\Modules;C:\\\\\\\\Program Files\\\\\\\\WindowsPowerShell\\\\\\\\Modules;C:\\\\\\\\Windows\\\\\\\\system32\\\\\\\\WindowsPowerShell\\\\\\\\v1.0\\\\\\\\Modules;C:\\\\\\\\Program Files (x86)\\\\\\\\AWS Tools\\\\\\\\PowerShell\\\\\\\\ PUBLIC=C:\\\\\\\\Users\\\\\\\\Public SystemDrive=C: SystemRoot=C:\\\\\\\\Windows TEMP=C:\\\\\\\\Users\\\\\\\\ADMINI~1\\\\\\\\AppData\\\\\\\\Local\\\\\\\\Temp TMP=C:\\\\\\\\Users\\\\\\\\ADMINI~1\\\\\\\\AppData\\\\\\\\Local\\\\\\\\Temp USERDOMAIN=EC2AMAZ-3QHKPN9 USERDOMAIN_ROAMINGPROFILE=EC2AMAZ-3QHKPN9 USERNAME=Administrator USERPROFILE=C:\\\\\\\\Users\\\\\\\\Administrator windir=C:\\\\\\\\Windows WSLENV=WT_SESSION::WT_PROFILE_ID WT_PROFILE_ID={574e775e-4f2a-5b96-ac1e-a2962a402336} WT_SESSION=2a5ccf8c-1d0b-40b8-a291-6c07c5592132]\\\"\"\ntime=\"2023-10-12T21:12:43Z\" level=error msg=\"[hostagent] failed to convert path to mingw, maybe not using Git ssh?\"\ntime=\"2023-10-12T21:12:43Z\" level=warning msg=\"[hostagent] failed to run [ssh -V]: stderr=\\\"\\\"\"\ntime=\"2023-10-12T21:12:43Z\" level=error msg=\"[hostagent] failed to convert path to mingw, maybe not using Git ssh?\"\ntime=\"2023-10-12T21:12:43Z\" level=info msg=\"[hostagent] hostagent socket created at C:\\\\Users\\\\Administrator\\\\Code\\\\finch\\\\_output\\\\lima\\\\data\\\\finch\\\\ha.sock\"\ntime=\"2023-10-12T21:12:43Z\" level=info msg=\"[hostagent] Starting WSL VM\"\ntime=\"2023-10-12T21:12:43Z\" level=warning msg=\"[hostagent] wsl2 driver is experimental and seems currently unstable. Help wanted to fix it.\"\ntime=\"2023-10-12T21:12:43Z\" level=info msg=\"[hostagent] Attempting to download the image\"\ntime=\"2023-10-12T21:12:44Z\" level=info msg=\"[hostagent] decompressing .gz with [gzip -d]\"\ntime=\"2023-10-12T21:12:44Z\" level=info msg=\"[hostagent] Decompressing the image (finch-rootfs-production-amd64-1696963702.tar.gz)\\n\"\n2023/10/12 21:12:44 Error in Watcher Error channel: CancelIo: The handle is invalid.\n2023/10/12 21:12:44 Error in Watcher Error channel: CloseHandle: The handle is invalid.\ntime=\"2023-10-12T21:12:44Z\" level=fatal msg=\"exiting, status={Running:false Degraded:false Exiting:true Errors:[] SSHLocalPort:0} (hint: see \\\"C:\\\\\\\\Users\\\\\\\\Administrator\\\\\\\\Code\\\\\\\\finch\\\\\\\\_output\\\\\\\\lima\\\\\\\\data\\\\\\\\finch\\\\\\\\ha.stderr.log\\\")\"\n"
time="2023-10-12T21:12:44Z" level=fatal msg="exit status 1"
```